Algorithm.CSharp/ClassicRangeConsolidatorAlgorithm.cs

```diff
@@ -0,0 +1,42 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Data.Consolidators;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Example algorithm of how to use the ClassicRangeConsolidator
+    /// </summary>
+    public class ClassicRangeConsolidatorAlgorithm : RangeConsolidatorAlgorithm
```

Comment:
Could this one assert that when price variation is high enough, we don't get the intermediate bars?

Reply:
It indeed does! In `OnDataConsolidated()` method, after calling the base implementation of this method it checks all the bars have non-zero volume since intermediate/phantom bars have zero volume. However, thanks for reminding me to write a comment in `RangeConsolidator.cs` explaining this property of intermediate/phantom RangeBar's 💯